### PR TITLE
Added friends app

### PR DESCRIPTION
**Pull request by SerhiiYakovenko**
# Friends app task.

[Demo](https://serhiiyakovenko.github.io/friends-app/)  |  [Code](https://github.com/SerhiiYakovenko/friends-app/)

The code is submitted in a dedicated feature branch.
Please, review.

### CODE REVIEW

File: submissions/SerhiiYakovenko/friends-app/script.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+fetch(API_URL)
+  .then((response) => {
+    return response.json();
+  })
+  .then(function (json) {
+    users = json.results;
+    render(users);
+  })
+  .catch((error) => console.log(error));
```
Comment (zonzujiro) [inline submissions/SerhiiYakovenko/friends-app/script.js:19]: Put into some function with meaningful name

File: submissions/SerhiiYakovenko/friends-app/script.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  main.appendChild(usersCards);
+}
+
+function sortAgeAsc(users) {
+  createCards(users.sort((a, b) => a.dob.age - b.dob.age));
+}
+
+function sortAgeDesc(users) {
+  createCards(users.sort((a, b) => b.dob.age - a.dob.age));
+}
```
Comment (zonzujiro) [inline submissions/SerhiiYakovenko/friends-app/script.js:44]: The sort function should do only sorting.

Also, it should be one function. The only order of arguments matters

File: submissions/SerhiiYakovenko/friends-app/script.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+function sortNameDesc(users) {
+  createCards(
+    users.sort((a, b) => {
+      if (b.name.first < a.name.first) return -1;
+      if (b.name.first > a.name.first) return 1;
+      return 0;
+    })
+  );
+}
```
Comment (zonzujiro) [inline submissions/SerhiiYakovenko/friends-app/script.js:64]: One function

File: submissions/SerhiiYakovenko/friends-app/script.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  });
+  createCards(searchResult);
+}
+
+function render(users) {
+  createCards(users);
+  ageAscending.addEventListener("click", () => sortAgeAsc(users));
+  ageDescending.addEventListener("click", () => sortAgeDesc(users));
+  nameAscending.addEventListener("click", () => sortNameAsc(users));
+  nameDescending.addEventListener("click", () => sortNameDesc(users));
```
Comment (zonzujiro) [inline submissions/SerhiiYakovenko/friends-app/script.js:84]: Please, use event delegation.
Also, you can listen for `change` event.

**Review by zonzujiro — CHANGES_REQUESTED** [review]

File: submissions/SerhiiYakovenko/friends-app/index.html [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      <h1>Free friends without registration and SMS</h1>
+    </header>
+    <aside>
+      <input class="search" type="text" placeholder="Search for a friend" />
+      <span class="sort-by_age">Sort by Age</span>
+      <span class="age-desc">&#9660;</span>
+      <span class="age-asc">&#9650;</span>
+      <span class="sort-by_name">Sort by Name</span>
+      <span class="name-desc">&#9660;</span>
+      <span class="name-asc">&#9650;</span>
```
Comment (zonzujiro) [inline submissions/SerhiiYakovenko/friends-app/index.html:25]: Because of a11y this also should be inputs.

**Comment by SerhiiYakovenko** [comment]
Thanks for the advice @zonzujiro, please review 🥺

File: submissions/SerhiiYakovenko/friends-app/script.js [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+    ? users.sort((a, b) => a.dob.age - b.dob.age)
+    : users.sort((a, b) => b.dob.age - a.dob.age);
+}
+
+function sortByName(users, order) {
+  return users.sort((a, b) => {
+    if (order.includes("asc")) {
+      return a.name.first < b.name.first ? -1 : 1
+    } else {
+      return b.name.first < a.name.first ? -1 : 1
```
Comment (zonzujiro) [inline submissions/SerhiiYakovenko/friends-app/script.js:51]: Same function as at previous line. DRY

File: submissions/SerhiiYakovenko/friends-app/script.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+      searchResult.push(user);
+    }
+  });
+  return searchResult;
+}
+
+function render(users) {
+  createCards(users);
+  ageAscending.addEventListener("change", (event) =>
+    createCards(sortByAge(users, event.target.className))
```
Comment (zonzujiro) [inline submissions/SerhiiYakovenko/friends-app/script.js:72]: Please, use class names only for styling. For your purposes you can choose `data-*`

**Review by zonzujiro — CHANGES_REQUESTED** [review]

**Comment by SerhiiYakovenko** [comment]
Updated, @zonzujiro can you please take a look?

**Review by zonzujiro — APPROVED** [review]
LGTM

**Pull request merged**